src/App/App.js

```diff
@@ -83,20 +82,16 @@ export class App extends Component{
     } 
   }
 
-   addRating = (desiredRating) => {
-    let usersRating = this.state.ratedMovies.find(ratedMovie => ratedMovie.movie_id === this.state.movieID)
-    if (!desiredRating) {
+   addRating = async (desiredRatingByUser) => {
+    const movieToRate = this.state.ratedMovies.find(ratedMovie => ratedMovie.movie_id === this.state.movieID)
+    if (!desiredRatingByUser) {
       return alert('You have to select a rating to rate a movie!')
-    } else if (!usersRating) {
-    let newRating = {movie_id: this.state.movieID, rating: desiredRating.value}
-      fetcher.fetchCreateUserRating(this.state.userData.id, newRating)
-      .then(() => fetcher.fetchUserRatings(this.state.userData.id)
-      // .then(() => fetcher.fetchAllMovies()
-      // .then(promise => this.setState({movies: promise.movies}))
-      // )
-      // .then(() => this.componentDidMount())
-      .then(ratedMovies => this.setState({ ratedMovies }))
-      )
+    } else if (!movieToRate) {
+      const newRating = {movie_id: this.state.movieID, rating: desiredRatingByUser.value}
+      const addNewRating = await fetcher.fetchCreateUserRating(this.state.userData.id, newRating)
+      const allUserRatings = await fetcher.fetchUserRatings(this.state.userData.id)
+      this.getMovieDetails(this.state.movieID)
+      this.setState({ratedMovies: allUserRatings})
```

Comment:
this.setState( {ratedMovies: allUserRatings} ) for readability! Also, we'll have to go back and lint the entire project. I know that is part of our grading rubric.

Reply:
oh shooot!!